### ISSUE TITLE
NVIDIA GeForce RTX 4070 Ti Memory Junction reports 0C all the time

### ISSUE THREAD

**Issue by Shineblind** [TimSirmovics/FanControl.NvThermalSensors#2]
Hey, 

my NVIDIA GeForce RTX 4070 Ti Memory Junction sensor reports 0C all the time which obviously wont trigger any fan curves.
Hot Spot seems to be working fine though,

is this function supported at all for 40 series cards?

**Comment by TimSirmovics** [comment]
I haven't personally tested with a 4070Ti.
Could you share your log.txt file?

**Comment by Shineblind** [comment]
[log.txt](https://github.com/user-attachments/files/15514332/log.txt)

Im not sure the last time the log file was saved, I tried enabling some stuff and disabling it, I send a screenshot of what sensors I got checked as well currently
![image](https://github.com/TimSirmovics/FanControl.NvThermalSensors/assets/17359901/e3f85a3a-888f-46dd-8ee4-32f7ae95105f)

**Comment by TimSirmovics** [comment]
What about if you try HWiNFO64, does that show the memory junction sensor?

**Comment by Shineblind** [comment]
yes, standalone (not linked to FanControl somehow, iirc you need HWiNFO64 premium for that)
I can see memory junction temp as well as in GPU-Z

**Comment by TimSirmovics** [comment]
Can you run the executable in the attached zip file, copy the console output and reply with it here.
Also monitor your memory junction sensor at the same time, and note whether any of the temp values displayed in the console match up.

[FanControl.NvThermalSensors.Debug.zip](https://github.com/user-attachments/files/15804262/FanControl.NvThermalSensors.Debug.zip)

**Comment by changhexiong** [comment]
Same problem,but RTX 4070 Ti SUPER.
HWiNFO64 and AIDA64 can report the correct memory junction.

Debug outputs here.
I think the last column is the correct memory junction.

`Found 1 GPU

GPU 1: NVIDIA GeForce RTX 4070 Ti SUPER
Mask: 11111111000000000000000000000000
Temps 0:    43.08   52.94   46.97   46.47   45.94   52.94   46.00   52.00
Temps 1:    42.92   52.84   46.78   46.31   45.84   52.84   46.00   50.00
Temps 2:    43.03   52.69   46.91   46.41   45.69   52.69   46.00   52.00
Temps 3:    42.81   52.66   46.62   46.16   45.66   52.66   46.00   52.00
Temps 4:    42.76   52.62   46.50   46.09   45.62   52.62   46.00   50.00
Temps 5:    45.86   53.66   51.47   49.72   46.66   53.66   47.00   52.00
Temps 6:    47.81   54.66   54.66   52.00   47.62   54.62   48.00   52.00
Temps 7:    44.36   53.78   48.72   47.97   46.78   53.78   47.00   52.00
Temps 8:    46.93   54.53   53.34   50.97   47.53   54.53   48.00   54.00
Temps 9:    49.52   57.31   57.31   54.00   47.91   54.91   48.00   54.00

Finished.
`

**Comment by TimFx7** [comment]
Same problem,
I checked with aida64. I think the last column is the correct memory junction. (%100)

GPU 1: NVIDIA GeForce RTX 4090
Mask: 11111111000000000000000000000000
Temps 0:    34,76   42,16   37,94   35,97   35,16   42,16   34,00   38,00
Temps 1:    35,25   42,50   38,31   36,50   35,50   42,50   35,00   38,00
Temps 2:    35,42   42,56   38,62   36,69   35,56   42,56   35,00   38,00
Temps 3:    35,51   42,75   38,66   36,78   35,75   42,75   35,00   38,00
Temps 4:    35,56   42,88   38,81   36,84   35,88   42,88   35,00   38,00
Temps 5:    35,62   42,91   38,84   36,91   35,91   42,91   35,00   38,00
Temps 6:    34,68   42,41   37,62   35,88   35,41   42,41   35,00   38,00
Temps 7:    34,36   42,31   37,34   35,53   35,31   42,31   34,00   38,00
Temps 8:    34,14   42,19   37,19   35,28   35,19   42,19   34,00   36,00
Temps 9:    33,99   42,00   36,78   35,12   35,00   42,00   34,00   36,00

Finished.

**Comment by TimSirmovics** [comment]
Can you please try this updated debug app and let me know if the sensor values are correct.

[FanControl.NvThermalSensors.Debug.zip](https://github.com/user-attachments/files/19161421/FanControl.NvThermalSensors.Debug.zip)

**Comment by TimFx7** [comment]
> Can you please try this updated debug app and let me know if the sensor values are correct.
> 
> [FanControl.NvThermalSensors.Debug.zip](https://github.com/user-attachments/files/19161421/FanControl.NvThermalSensors.Debug.zip)

@TimSirmovics 
Sensor values are exactly correct, I checked with gpu-z software.

**Comment by changhexiong** [comment]
> Can you please try this updated debug app and let me know if the sensor values are correct.
> 
> [FanControl.NvThermalSensors.Debug.zip](https://github.com/user-attachments/files/19161421/FanControl.NvThermalSensors.Debug.zip)

correct, checked with HWiNFO64

**Comment by TimSirmovics** [comment]
Please try the latest release.

**Comment by TimFx7** [comment]
> Please try the latest release.

unfortunately the same problem persists.

**Comment by TimSirmovics** [comment]
Sorry can you grab the dll from the v1.2 release again.

**Comment by TimFx7** [comment]
> Sorry can you grab the dll from the v1.2 release again.


working now thanks. 🥇

**Comment by d-e-n-n-i-s-t** [comment]
The latest release also fixed VRAM temp reading on my 4060Ti (MSI Ventus 2x)

**Comment by d-e-n-n-i-s-t** [comment]
> The latest release also fixed VRAM temp reading on my 4060Ti (MSI Ventus 2x)

@TimSirmovics I was wondering, is this even possible? Ever since 1.2, I get a readout for memory junction also. But before, I didn't. I guess I'm just curious whether this readout is real or buggy.
I started gpu stress tests and I _think_ the sensor reports coherent values

**Comment by TimSirmovics** [comment]
As far as I can find, there is no official program from Nvidia that displays this this temperature.
The best you can do the verify it's accuracy is to compare against other hardware monitoring programs (HWiNFO64, GPU-Z, AIDA64 etc.)

**Comment by d-e-n-n-i-s-t** [comment]
@TimSirmovics That's the interesting part, no other program has ever reported me any memory junction temp (HWiNFO64, LHM) before. That's why I assumed my graphics card has no temp sensors for VRAM.
But I'm very sure the value that gets reported since 1.2 is real and accurate. It gets a little warm when starting a VRAM stress test (OCCT) and gets even warmer when GPU is stressed (heat soak)

The only question is, why does LHM not detect this sensor? I saw you created a pull request some years ago that should enable LHM to show this temp.

**Comment by TimSirmovics** [comment]
If that is the case then the value may not be accurate for your card.
The temperature sensor values for the Hot Spot and VRAM as I mentioned are undocumented and there is no official source to compare to for knowing exactly where they are held in memory.
